### ISSUE TITLE
[BUG]: Magic getters for relationships throws exception when there is no relation in 4.0

### ISSUE THREAD

**Issue by wurst-hans** [phalcon/cphalcon#14460]
My projects are running on Phalcon 3.4. There I'm using many model relationships like

```
class Requests extends Model {
    public function initialize() {
        $this->belongsTo('customerId', 'App\Models\Customers', 'id', [ 'alias' => 'Customer' ]);
	}
}
```

which works fine in Phalcon 3.4, so I can use it like

```
$request = Requests::findFirst();
$customer = $request->getCustomer();
```

At the moment, I'm trying to migrate all projects to Phalcon 4.0.0.-rc.1-783 but when executing above code, I get an error

```
Phalcon\Mvc\Model\Exception: The method 'getCustomer' doesn't exist on model 'App\Models\Requests' in ...
```

I found out that `getCustomer()` fails and throws exception if `customerId` is `NULL` in model `Requests`. In Phalcon 3.4 I get a `false` as result, but 4.0 throws an exception. 
Getting `false` allows to initialize a new object, when there is no other already. For this I can modify the code to

```
$request = Requests::findFirst();
$customer = $request->getCustomer();
$customer = $customer ?: new Customers();
```

There are hundreds of lines of code in my projects which have to be altered then to prevent this exception in Phalcon 4.0. Is there a workaround or is it planned to add 3.4 behavior again?

IMHO: It's not correct to not add magic getters when column used for relationship is `NULL`. The expected behavior (like when calling `findFirst()` without result) is getting `false` or an object.

Forgot to mention, that I have tried it with lowercase alias already without success. Even with Vokuro this happens.

**Comment by Jurigag** [comment]
Can you check what happens if you add `allowNulls => true` ?

**Comment by wurst-hans** [comment]
> Can you check what happens if you add `allowNulls => true` ?

You mean by adding

```
'foreignKeys' => [
    'allowNulls' => true
],
```

to model `Requests`? Nothing changed, exception is still thrown.

**Comment by wurst-hans** [comment]
> @wurst-hans, this has just been fixed in #14444
> Please test the current 4.0.x branch if you can.

Looks good. No exception is thrown and I get `NULL` for non existing relationship. Nevertheless, one could prevent unnecessary database request when relationship key is `NULL`.
Anyway, thanks for your help. When will this be available in mainline repo?

**Comment by zsilbi** [comment]
> Nevertheless, one could prevent unnecessary database request when relationship key is `NULL`.

Please feel free to create a NFR in another issue. 👍 

> Anyway, thanks for your help. When will this be available in mainline repo?

It will be included in the following 4.0.0-rc.2 release in the next few weeks.

**Comment by ruudboon** [comment]
@wurst-hans Can you open a new issue for the suggested improvement (good catch!)? Don't want to log this in the same issue. Closing this one as duplicate.

**Comment by apoca** [comment]
> > @wurst-hans, this has just been fixed in #14444
> > Please test the current 4.0.x branch if you can.
> 
> Looks good. No exception is thrown and I get `NULL` for non existing relationship. Nevertheless, one could prevent unnecessary database request when relationship key is `NULL`. Anyway, thanks for your help. When will this be available in mainline repo?

Well, I'm trying to upgrade from v3.4 to v5 and I have the same issue in Phalcon v5. My doubt is "This bug is also fixed in version v5"?
